src/main/java/carRace/step5/domain/CarRacingGame.java

```diff
@@ -1,99 +1,79 @@
-
-        for (int i = 0; i < lastGameList.size(); i++) {
-            selectWinner(maxValue, winnerList, lastGameList.get(i));
-        }
-
-        System.out.print("우승자 : " + winnerList.get(0).getCarName());
-        for (int i = 1; i < winnerList.size(); i++) {
-            System.out.print(", " + winnerList.get(i).getCarName());
-        }
+    public void RunPlayGame(String[] carNames) {
+        makeCars(carNames);
+        playRasingGame();
+        drawRasingWinner();
     }
 
-    public void drawRasingGame() {
+    private void playRasingGame() {
         for (int i = 0; i < raceTryCount; i++) {
-            for (String carName : carNames) {
-                System.out.print(carName + " : "
-                 + CarRasingDraw.changeNumberToChar(carResultData.get(carName).get(i).getMoveDistance()));
-                System.out.print(System.lineSeparator());
-            }
+            repeatRaceTryCount();
             System.out.print(System.lineSeparator());
         }
     }
 
-    private void repeatRaceTryCount(List<Car> paramList) {
-        Car car;
-        String carName = paramList.get(0).getCarName();
-
-        for (int i = 1; i < raceTryCount; i++) {
-            car = new Car(paramList.get(i - 1));
-            car.moveCar(random.nextInt(10));
-            paramList.add(car);
+    private void repeatRaceTryCount() {
+        for (int i = 0; i < carsScore.size(); i++) {
+            carsScore.get(i).moveCar(random.nextInt(10));
+            CarRasingDraw.printCarDistance(carsScore.get(i).getCarName(), carsScore.get(i).getMoveDistance());
         }
-        this.carResultData.put(carName, paramList);
     }
 
-    private List<Car> getLastPlay() {
-        List<Car> lastGameList = new ArrayList<>();
+    private void drawRasingWinner() {
+        int maxScore = getMaxWinnerSocre();
+        List<String> winnerList = new ArrayList<>();
 
-        for (String carName : carNames) {
-            lastGameList.add(carResultData.get(carName).get(raceTryCount - 1));
+        for (int i = 0; i < carsScore.size(); i++) {
+            selectWinner(maxScore, winnerList, i);
         }
 
-        return lastGameList;
+        CarRasingDraw.printWinner(winnerList);
+    }
+
+    private void selectWinner(int maxValue, List<String> winnerList, int index) {
+        if (carsScore.get(index).getMoveDistance() >= maxValue) {
+            winnerList.add(carsScore.get(index).getCarName());
+        }
     }
 
-    private int getMaxWinnerSocre(List<Car> lastGameList) {
+    private int getMaxWinnerSocre() {
         int maxValue = 0;
 
-        for (int i = 0; i < lastGameList.size(); i++) {
-            if (maxValue <= lastGameList.get(i).getMoveDistance()) {
-                maxValue = lastGameList.get(i).getMoveDistance();
+        for (int i = 0; i < carsScore.size(); i++) {
+            if (maxValue <= carsScore.get(i).getMoveDistance()) {
+                maxValue = carsScore.get(i).getMoveDistance();
```

Comment:
다음과 같이 depth를 줄여볼 수 있을 것 같아요 :)
```suggestion
            maxValue = Math.max(carsScore.get(i).getMoveDistance(), maxValue);
```